### PR TITLE
feat: collapsed cadp

### PR DESCRIPTION
**Pull request by JeanPierreNell**
In Relation to : [Remove the CADProc from the default transaction evaluation flow#6] (https://github.com/frmscoe/General-Issues/issues/6)

### CODE REVIEW

File: src/services/helper.service.ts [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+      id: channel.id,
+      typologyResult: typologyResults,
+    };
+    const apmTadProc = apm.startSpan('tadProc.exec');
+
+    const message = networkMap.messages.find((tran) => tran.txTp === transaction.TxTp);
+    const channelResults = await handleChannels(message!, transaction, networkMap, channelResult);
+
+    apmTadProc?.end();
+    span = apm.startSpan(`[${transactionID}] Delete Typology interim cache key`);
```
Comment (johanfol) [inline src/services/helper.service.ts:151]:
```suggestion
    span = apm.startSpan(`[${transactionID}] Delete Channel interim cache key`);
```

File: src/services/helper.service.ts [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+  const span = apm.startSpan('handleChannels');
+
+  try {
+    const transactionType = 'FIToFIPmtSts';
+    const transactionID = transaction[transactionType].GrpHdr.MsgId;
+
+    // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+    const cacheKey = `tadp_${transactionID}_${message.id}_${message.cfg}`;
+    const spanDBMembers = apm.startSpan('db.get.members');
+    const jtypologyCount = await databaseManager.addOneGetCount(cacheKey, JSON.stringify(channelResult));
```
Comment (johanfol) [inline src/services/helper.service.ts:27]: This should be jChannelCount, not typology. It's misleading.

File: __tests__/unit/logic.test.ts [hunk 3]
```diff
@@ -193,10 +224,12 @@ describe('TADProc Service', () => {
         const channelResult = requestBody.channelResult;
         const message = networkMap.messages.find((tran) => tran.txTp === transaction.TxTp);
 
-        const result = await handleExecute(requestBody);
-        expect(result).toBeDefined();
-        if (res?.report?.status) expect(res.report.status).toBe('ALRT');
-        else throw 'Test failed - expect response to be called';
+        if (message) {
```
Comment (johanfol) [inline __tests__/unit/logic.test.ts:227]: bad practice to put this guard in unit tests - if someone modifies the method that returns the networkmap to return null, the tests will no longer run and not report an error / failure of the test to run. 

If you need to add the null check, at least put a throw in the "else" of the guard. That would be fine.

**Review by johanfol — CHANGES_REQUESTED** [review]

**Review by johanfol — APPROVED** [review]

**Comment by JeanPierreNell** [comment]
Pulled in Protobuff changes per: https://github.com/frmscoe/General-Issues/issues/265

**Review by cshezi — APPROVED** [review]
🐐 ed

**Review by rtkay123 — APPROVED** [review]

**Pull request merged**